### PR TITLE
feat: add new Go bindings to perform HTTP requests

### PR DESCRIPTION
**Pull request by Angelmmiguel**
Define the new HTTP bindings for the Go kit and expose a new `SendHttpRequest` method. It includes new WIT files that are required due to a mismatch between the `wit-bindgen` version used by the host (v0.2) and the Go kit (v0.6). I also had to adapt the generated code to work between these versions. You have all the information in the `kits/go/worker/README.md` file.

## Changes

- Define new WIT files that outputs the same bindings as `wit/core` but are compatible with `wit-bindgen` v0.6.
- Generate the bindings following the approach described on the `kits/go/worker/README.md` file.
- Add a new `SendHttpRequest` method in the Go kit.
- Ensure this method interacts with the `http.Request` and `http.Response` structs instead of the internal ones.
- Add new examples and update the documentation so current ones work properly Wasm Workers Server v1.3.0

It refs #165

### CODE REVIEW

File: kits/go/worker/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Wasm Workers Server is on the road to adopt Wasm components, but it's not there yet. However, we started adopting WIT to generate the bindings for the different languages.
+
+The host (Wasm Workers Server) and other languages like Rust and JavaScript rely on [wit-bindgen v0.2](https://github.com/bytecodealliance/wit-bindgen/tree/v0.2.0). However, the Go bindings were not available on that version so it caused some extra work to generate the Go bindings.
+
+These are the steps to recreate the current Go bindings:
+
+- Clone the wit-binding repository and checkout to the [35cb45f2](https://github.com/bytecodealliance/wit-bindgen/commit/35cb45f25eb113b54406f269778d46a37716a7c5) commit (between v0.6 - v0.7). This commit produces compatible binding identifiers and fixes an error with the types on the generated C / Go code:
+
+    ```
```
Comment (ereslibre) [inline kits/go/worker/README.md:15]:
```suggestion
    ```shell-session
```

**Review by ereslibre — COMMENTED** [review]
LGTM, great work @Angelmmiguel. Minor comments and questions.

File: wit/go-ephemeral/http-types.wit
[the remark this answers is not in the document]
Comment (Angelmmiguel) [reply wit/go-ephemeral/http-types.wit:1]: Exactly. WIT has evolved a lot and there're some breaking changes while the spec stabilizes. The end goal is to have a single WIT definition once we can recreate all kits based on it. 

Currently, this is the situation:

* core: oldest WIT version
* go-ephemeral: WIT version before packages

**Review by ereslibre — APPROVED** [review]
LGTM 👏

**Pull request merged**